core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -288,13 +288,36 @@ class half_t {
     return *this;
   }
 
+  template <class T>
+  KOKKOS_FUNCTION void operator=(T rhs) volatile {
```

Comment:
Any reason to return `void` rather than `half_t volatile &`?

Reply:
To suppress "warning: implicit dereference will not access object of type ‘volatile __half’ in statement". See https://github.com/kokkos/kokkos/commit/9f23b5f9b14df1d9d13f7c65a9c99b3fae6bd410.